### ISSUE TITLE
My character didn't sleep in 10 days and is still kicking

### ISSUE THREAD

**Issue by Alapaga** [CleverRaven/Cataclysm-DDA#37344]
# Describe the bug

As simple as it sounds, drinking atomic coffee for a few days made my characte incapable of sleeping. Things i did before in attempting to tire my character include, digging and refilling pits, building a palissade gate for a couple days, taking a codeine pill, taking 2/3 ambien pills in the last few hours and drinking 2 hip flasks of strong alcohol and trying to sleep for a few times in the last 6 hours, and still no tired status.

# Steps To Reproduce

1. Make a character and give them the atomic coffee maker
2. Try to drink one unit of atomic coffee per meal for a week or so
3. You now achieved totally awake status for a few days now, you don't need sleeping anymore.
4. Stopping the consumption of stimulants for a few days didn't help in regaining the ability to sleep.

# Expected behavior

The main reason I want my character to sleep is to recover from his headshot wound.
I expected there to be a hard limit on how much time you can stay awake. In real life there are some experiments that were carried out to document how much time someone can stay awake and the limit should be around 11-12 days. I know there are some exceptions because of mutations, but my current character is not a mutant, he was a heavy sleeper from the start of the game. Also I suspect something strange to be happening in monster generation around where I am. Even tho i am wearing earplugs to isolate my character, i still get messages saying "something makes noise" in the log. I investigated the wails at an earlier date, and it seems that there's something going on underground, under a ranch of all things.
I know there's a discussion going on about this here 
https://github.com/CleverRaven/Cataclysm-DDA/issues/37069
This is my practical take on the subject. I hope this issue gets adressed at some point because sleeping is what makes us human. 

# Versions and configuration

- OS: Windows
    - OS Version: 10.0 1909
- Game Version: 0.D-11545-ge497763 [64-bit]
- Graphics Version: Tiles
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Generic Guns [generic_guns],
    Alternative Map Key [alt_map_key],
    Cars to Wrecks [cars_to_wrecks],
    Disable Religious Texts [no_religious_Texts],
    No Explosive Zombies [No_Explosive_Zombies],
    Classes and Scenarios Mod [more_classes_scenarios],
    Safe Autodoc [safeautodoc],
    SpeedyDex [speedydex],
    Stats Through Kills [stats_through_kills],
    Stats Through Skills [StatsThroughSkills]
]

# Additional context

[manual_backup5.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/4108746/manual_backup5.zip)
[debug.log](https://github.com/CleverRaven/Cataclysm-DDA/files/4108749/debug.log)

**Comment by Taichi22** [comment]
I'm not seeing a discussion at the provided link.
I'm aware of this implementation -- it acts this way because sleep is a simple int value that is either filled or not -- it has nothing to do with the real need to sleep, and, in effect, atomic coffee acts as if you slept, and then keeps you awake accordingly, though lowering your health in the process.
The only way I'm aware of countering this is through regeneration or fast healer-tree mutations, though I have been trying to think of a way to more realistically implement sleep requirements.

**Comment by TechyBen** [comment]
Could there not be a simple check for number of days without sleep and force it on the char if not met? Would seem reasonable (and a bit of a catch all for other unintended non-sleeping characters, though another catch all for mutations that prevent sleep I guess).

**Comment by Alapaga** [comment]
I corrected the provided link, and yes, it's a weird interaction to have with stimulants refilling the sleep metric. I have to admit I was not very carefull with the health stat calculations for this character since I used the simplified nutrition mod in this world gen. My character needs comfort food to keep himself going, and having cake helps him cope with all the horrible things he's witnessing everyday.
